src/components/sdk/MeetingControls/VideoInputBackgroundBlurControl.tsx

```diff
@@ -93,67 +75,74 @@ const VideoInputBackgroundBlurControl: React.FC<Props> = ({
       await selectVideoinput(current);
     } catch (error) {
       console.error('Failed to toggle Background Blur');
+    } finally {
+      setIsLoading(false);
     }
-
-    setIsLoading(false);
   };
 
-  const handleClick = async (deviceId: string): Promise<void> => {
-    try {
-      // If background blur is on, then re-use the same video transform pipeline, but replace the inner device
-      // If background blur is not on, then do a normal video selection
-      if (isVideoTransformDevice(activeVideoDevice) && !isLoading) {
-        setIsLoading(true);
-        if ('chooseNewInnerDevice' in activeVideoDevice) {
-          const transformedDevice =
-            // @ts-ignore
-            activeVideoDevice.chooseNewInnerDevice(deviceId);
-          await selectVideoinput(transformedDevice);
+  useEffect(() => {
+    const handleClick = async (deviceId: string): Promise<void> => {
```

Comment:
Now that you have moved `handleClick` inside `useEffect`, you have to also add methods or variables used in `handleClick` to dependency array. For e.g.: `selectVideoinput`